src/coordsim/metrics/metrics.py

```diff
@@ -49,6 +49,7 @@ def reset_run_metrics():
     metrics['run_avg_end2end_delay'] = 0.0
     metrics['run_max_end2end_delay'] = 0.0
     metrics['run_processed_flows'] = 0
+    metrics['run_max_node_usage'] = defaultdict(float)
```

Comment:
Minor thing: Could you please explicitly create a dict an initialize it with 0.0 for all nodes rather than using a defaultdict? Defaultdicts don't print well in the logs.

Something like `= {v: 0.0 for v in network.nodes.keys()}` or something should be sufficient.

Reply:
Sorry, I just got the notification for this comment along with the notification for the pull request. Is it urgent so that I can do it immediately or is it fine if it is done in the next PR? 
A very quick solution that I have is to just cast the defaultdict as dict in the `get_metrics()` function. This way we dont need to pass the network nodes to the metrics module.

Reply:
Ah, good idea with the `get_metrics()`. I'll take care of it in my next merge. So you don't have to worry about it